PanelViewController/Framework source/DraggableView.swift

```diff
@@ -36,14 +48,43 @@ class DraggableView: UIView {
         
         recognizer.setTranslation(CGPoint.zero, in: targetView)
         switch recognizer.state {
-        case .began:
-            delegate?.draggingBegan(view: self)
-        case .ended:
+        case .began, .changed:
+            if !(delegate?.shouldDrag(view: self, location: point) ?? true) {
+                // Delegate says we should not drag.
+                recognizer.isEnabled = false
+                return
+            }
+            if recognizer.state == .began {
+                delegate?.draggingBegan(view: self)
+            }
+        case .cancelled, .ended:
+            recognizer.isEnabled = true
             let velocity = recognizer.velocity(in: targetView)
             delegate?.draggingEnded(view: self, velocity: CGPoint(x: 0, y: velocity.y))
         default:
             // no op
             break
         }
     }
+    
+    override func point(inside point: CGPoint, with event: UIEvent?) -> Bool {
```

Comment:
Sorry, I don't understand this - how would the user set the header view to be pass through?

Reply:
`pointInside` is tricky; let's meet